include/avk/cpp_utils.hpp

```diff
@@ -1161,83 +1161,142 @@ namespace avk
 
 #pragma region swap + dispose and handle help functions
 
-	/**
-	*	This concept captures those objects which have a size() method, like vectors	*	
-	*/
+	/**  This concept captures those objects which have a size() method, like vectors
+	 */
 	template <typename T>
 	concept has_size = requires (T x)
 	{
 		x.size();
 	};
 
-	/**
-	*	This concept captures those objects that are explicitely convertible to boolean
-	*  
-	*	Note: most vulkan resources contain an explicit boolean converter
-	*/
+	/** This concept captures those objects that are explicitely convertible to boolean
+	 *
+	 *	Note: most vulkan resources contain an explicit boolean converter.
+	 */
 	template <class T>
 	concept boolean_convertible = requires(T(&f)()) {
 		static_cast<bool>(f());
 	};
 
-	/**
-	* Swaps aOld and aNew and handles the life time of aOld
-	*
-	* The purpose of the functions in this section swap and handle lifetime
-	* in a more clean structure in the source code.
-	*
-	* Additionally, when possible, lifetime_handler+swap is ignored if the
-	* rhs parameter contains no values.
-	*
-	* Be aware that inital value  of aOld  will be disposed of
-	*
-	* after the call, aOld's memory will contain initial aNew's value
-	*
-	* @param aNew				new Resource which will be moved to old Resource's place after operation
-	* @param aOld				old Resource object will be disposed of
-	* @param aLifeTimeHandler	Life time handler for aOld
-	*
-	*/
+	/** Swaps aRhs and aLhs and handles the life time of aRhs.
+	 *
+	 *  Be aware that inital content of aRhs will be disposed of.
+	 *  Ater the execution: 1) The reference aLhs will be in an unspecified state after the execution.
+	 *                      2) aRhs's memory will contain initial aLhs's value.
+	 *
+	 *  Be aware that inital value of aRhs will be passed on to aRhsLifeTimeHandler.
+	 *
+	 *  @param aLhs                 Left-hand side object, the content of which will be moved to aRhs's place after operation
+	 *  @param aRhs                 Right-hand side object, the contect of which to be disposed of.
+	 *  @param aRhsLifeTimeHandler	Life time handler for aRhs
+	 */
 	template<typename T, typename F>
-	void emplace_and_handle_previous(T& aNew, T&& aOld, F&& aLifeTimeHandler)
+	void swap_and_lifetime_handle_rhs(T& aLhs, T&& aRhs, F&& aRhsLifeTimeHandler)

```

Comment:
I thought, we agreed on `assign_and_dispose_rhs` (but actually, also `assign_and_lifetime_handle_rhs` would be fine fine)?
The problem I had was with "swap" because it might be confusing and therefore, I suggested "assign", s.t. you assign the `aRhs` to the `aLhs`.

I would have changed it by myself, but I didn't because I think the usages of this function in _Gears-Vk_  are still confusing... or maybe even wrong:

The _existing_ variable (member variable of a class) shall NOT be moved into this function as the `aRhs` parameter. E.g. the following:
```
			auto lifetimeHandlerLambda = [wnd](avk::renderpass&& rp) { wnd->handle_lifetime(std::move(rp)); };
		avk::swap_and_lifetime_handle_rhs(newRenderPass, std::move(mRenderpass), lifetimeHandlerLambda);
		avk::swap_and_lifetime_handle_rhs(newClearRenderPass, std::move(mClearRenderPass), 
```
The `mRenderpass`  and `mClearRenderPass`  variables are those that should stay in place => hence, passed in via the `aLhs`  parameter. I am actually wondering why this works (or does it?) because it would immediately dispose of the newly created renderpasses (i.e. `newRenderPass` and `newClearRenderPass`) and assign the old values, leaving `mRenderpass` and `mClearRenderPass` in moved-from state as well, wouldn't it?

Reply:
So just rename is fine?

I am not sure what you mean by` "The existing variable (member variable of a class) shall NOT be moved into this function as the aRhs parameter."`

From the way values are dealt with on move, the above seems all correct to me.

Regardless: The whole purpose of this function to not have the lengthy if:else everywhere in main swap chain code + avoid putting empty references in handler. IF the costs don't match the work it does, we can just inline that code.

Reply:
updated the function as discussed.